osm2gtfs/core/elements.py

```diff
@@ -0,0 +1,177 @@
+    variants of the same service line.
+
+    In OpenStreetMap this is usually represented as "route_master" relation.
+    In GTFS this is usually represented as "route".
+
+    """
+    route_id = attr.ib()
+
+    route_type = attr.ib(default=None)
+    route_desc = attr.ib(default=None)
+    route_color = attr.ib(default="#FFFFFF")
+    route_text_color = attr.ib(default="#000000")
+
+    # Related route variants
+    _itineraries = attr.ib(default=attr.Factory(list))
+
+    def __attrs_post_init__(self):
+        '''
+        Populates the object with information obtained from the tags
+        '''
+        # pylint: disable=unsupported-membership-test,unsubscriptable-object
+        if 'colour' in self.tags:
+            self.route_color = self.tags['colour']
+
+        # pylint: disable=unsupported-membership-test,unsubscriptable-object
+        if 'ref:colour_tx' in self.tags:
+            self.route_text_color = self.tags['ref:colour_tx']
+
+        # pylint: disable=unsupported-membership-test,unsubscriptable-object
+        if 'route_master' in self.tags:
+            self.route_type = self.tags['route_master'].capitalize()
+        elif 'route' in self.tags:
+            self.route_type = self.tags['route'].capitalize()
+        else:
+            self.route_type = "Bus"
+
+    def add_itinerary(self, itinerary):
+
+        if self.route_id != itinerary.route_id:
+            raise ValueError('Itinerary route ID (' +
+                             itinerary.route_id +
+                             ') does not match Line route ID (' +
+                             self.route_id + ')')
+        # pylint: disable=no-member
+        self._itineraries.append(itinerary)
+
+    def get_itineraries(self):
+        return self._itineraries
+
+
+@attr.s
+class Itinerary(Element):
+    """A public transport service itinerary.
+
+    It's a representation of a possible variant of a line, grouped together by
+    a Line object.
+
+    In OpenStreetMap this is usually represented as "route" relation.
+    In GTFS this is not exlicitly presented but used as based to create "trips"
+
+    """
+    route_id = attr.ib()
+    shape = attr.ib()
+
+    line = attr.ib(default=None)
+    fr = attr.ib(default=None)
+    to = attr.ib(default=None)
+    duration = attr.ib(default=None)
+
+    # All stop objects of itinerary
+    stops = attr.ib(default=attr.Factory(list))
+
+    def __attrs_post_init__(self):
+        '''
+        Populates the object with information obtained from the tags
+        '''
+        # pylint: disable=unsupported-membership-test,unsubscriptable-object
+        if 'from' in self.tags:
+            self.fr = self.tags['from']
+
+        # pylint: disable=unsupported-membership-test,unsubscriptable-object
+        if 'to' in self.tags:
+            self.to = self.tags['to']
+
+        # pylint: disable=unsupported-membership-test,unsubscriptable-object
+        if 'duration' in self.tags:
```

Comment:
`duration` is not a standard tag you will find in all route_master relations: it should be handled in the creators (just like the `travel_time` and `frequency` tags used in Accra.

Reply:
[duration](http://wiki.openstreetmap.org/wiki/Key:duration) is IMHO the standard tag.

Reply:
The more I think through it the more I follow your approach. Mainly because there has been so many different and valid ideas on how to handle it: `travel_time` or `duration` tag in OSM. In the other issue they also requested to define this in the schedule source. So, yes, will move it to the schedule creators. Thanks for rising it.